lib/ecto/changeset.ex

```diff
@@ -1106,6 +1118,26 @@ defmodule Ecto.Changeset do
     end
   end
 
+  defp decimal_compare(result, :less_than) do
+    Decimal.equal?(result, Decimal.new(-1))
+  end
+
+  defp decimal_compare(result, :greater_than) do
+    Decimal.equal?(result, Decimal.new(1))
+  end
+
+  defp decimal_compare(result, :equal_to) do
+    Decimal.equal?(result, Decimal.new(0))
+  end
+
+  defp decimal_compare(result, :less_than_or_equal_to) do
+    decimal_compare(result, :less_than) || decimal_compare(result, :equal_to)
+  end
+
+  defp decimal_compare(result, :greater_than_or_equal_to) do
+    decimal_compare(result, :greater_than) || decimal_compare(result, :equal_to)
```

Comment:
`or` signals that you are working with booleans - not truish and falseish values. It is also slightly faster.